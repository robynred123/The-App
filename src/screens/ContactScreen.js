import React, { useState } from 'react'
import {
  Text,
  View,
  TextInput,
  StyleSheet,
  TouchableOpacity,
  Alert
} from 'react-native'
import validator from 'validator'

const ContactScreen = () => {
  const [name, setName] = useState('')
  const [email, setEmail] = useState('')
  const [message, setMessage] = useState('')
  const [validationError, setValidationError] = useState(null)

  const handleOnSubmit = () => {
    let isValid = validateValues(name, email)
    if (isValid) {
      if (validationError === null) {
        return (
          Alert.alert(
            'Message Submitted',
            `Name: ${name}, Email: ${email}, Message: ${message}`
          ),
          setName(''),
          setEmail(''),
          setMessage('')
        )
      }
    }
  }

  const validateValues = (name, email) => {
    setValidationError(null)

    if (!validator.isLength(name, { min: 5, max: undefined })) {
      setValidationError('Please Enter a Name')
      return false
    } else if (!validator.isEmail(email)) {
      setValidationError('Please Enter a Valid Email')
      return false
    }
    return true
  }

  return (
    <View style={styles.container}>
      <Text style={styles.title}>Contact Me</Text>
      <View style={styles.form}>
        <Text style={styles.text}>Name</Text>
        <TextInput
          style={styles.input}
          onChangeText={(text) => setName(text)}
          value={name}
        />
        <Text style={styles.text}>Email</Text>
        <TextInput
          style={styles.input}
          onChangeText={(text) => setEmail(text)}
          value={email}
        />
        <Text style={styles.text}>Message</Text>
        <TextInput
          style={styles.input}
          onChangeText={(text) => setMessage(text)}
          value={message}
          maxLength={500}
          multiline={true}
        />
        {validationError ? (
          <Text style={styles.error}>{validationError}</Text>
        ) : null}
        <View>
          <TouchableOpacity onPress={handleOnSubmit} style={styles.button}>
            <Text>Submit</Text>
          </TouchableOpacity>
        </View>
      </View>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center'
  },
  form: {
    flexDirection: 'column',
    padding: 15,
    alignSelf: 'stretch',
    alignItems: 'center'
  },
  title: {
    fontSize: 20,
    fontWeight: '700',
    justifyContent: 'center',
    padding: 5,
    textAlign: 'center'
  },
  input: {
    backgroundColor: '#D1D1D1',
    borderRadius: 10,
    width: '80%',
    padding: 10,
    marginBottom: 30
  },
  text: {
    fontSize: 15,
    flexWrap: 'wrap'
  },
  button: {
    backgroundColor: '#AA7DC3',
    justifyContent: 'center',
    alignItems: 'center',
    padding: 15,
    paddingHorizontal: 25,
    borderRadius: 15
  },
  error: {
    fontSize: 15,
    flexWrap: 'wrap',
    color: 'red'
  }
})

export default ContactScreen
