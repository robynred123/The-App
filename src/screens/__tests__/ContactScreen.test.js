import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import ContactScreen from '../ContactScreen'

Enzyme.configure({ adapter: new Adapter() })

describe('<ContactScreen />', () => {
  const component = shallow(<ContactScreen />)

  it('should render the ContactScreen without crashing', () => {
    expect(component).toMatchSnapshot()
  })

  it('should render the ContactScreen with a title', () => {
    expect(component.find('Text').first().children().text()).toEqual(
      'Contact Me'
    )
    expect(component.find('Text').first().props().style).toEqual({
      fontSize: 20,
      fontWeight: '700',
      justifyContent: 'center',
      padding: 5,
      textAlign: 'center'
    })
  })

  it('should render the ContactScreen with a name TextInput', () => {
    expect(component.find('Text').at(1).children().text()).toEqual('Name')
    //expect(component.find('TextInput').at(1).props().value).toEqual('name')
  })

  it('should render the ContactScreen with an email TextInput', () => {
    expect(component.find('Text').at(2).children().text()).toEqual('Email')
    //expect(component.find('TextInput').at(2).props().value).toEqual('email')
  })

  it('should render the ContactScreen with a message TextInput', () => {
    expect(component.find('Text').at(3).children().text()).toEqual('Message')
    //expect(component.find('TextInput').at(2).props().value).toEqual('email')
  })
})
