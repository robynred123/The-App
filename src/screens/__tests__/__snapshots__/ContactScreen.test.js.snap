// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactScreen /> should render the ContactScreen without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "700",
        "justifyContent": "center",
        "padding": 5,
        "textAlign": "center",
      }
    }
  >
    Contact Me
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "flexDirection": "column",
        "padding": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "flexWrap": "wrap",
          "fontSize": 15,
        }
      }
    >
      Name
    </Text>
    <Component
      allowFontScaling={true}
      onChangeText={[Function]}
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "#D1D1D1",
          "borderRadius": 10,
          "marginBottom": 30,
          "padding": 10,
          "width": "80%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <Text
      style={
        Object {
          "flexWrap": "wrap",
          "fontSize": 15,
        }
      }
    >
      Email
    </Text>
    <Component
      allowFontScaling={true}
      onChangeText={[Function]}
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "#D1D1D1",
          "borderRadius": 10,
          "marginBottom": 30,
          "padding": 10,
          "width": "80%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <Text
      style={
        Object {
          "flexWrap": "wrap",
          "fontSize": 15,
        }
      }
    >
      Message
    </Text>
    <Component
      allowFontScaling={true}
      maxLength={500}
      multiline={true}
      onChangeText={[Function]}
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "#D1D1D1",
          "borderRadius": 10,
          "marginBottom": 30,
          "padding": 10,
          "width": "80%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View>
      <ForwardRef
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#AA7DC3",
            "borderRadius": 15,
            "justifyContent": "center",
            "padding": 15,
            "paddingHorizontal": 25,
          }
        }
      >
        <Text>
          Submit
        </Text>
      </ForwardRef>
    </View>
  </View>
</View>
`;
